lib/matplotlib/tests/test_offsetbox.py

```diff
@@ -182,3 +186,59 @@ def test_get_packed_offsets_equal(wd_list, total, sep, expected):
 def test_get_packed_offsets_equal_total_none_sep_none():
     with pytest.raises(ValueError):
         _get_packed_offsets([(1, 0)] * 3, total=None, sep=None, mode='equal')
+
+
+@pytest.mark.parametrize('child_type', ['draw', 'image', 'text'])
+@pytest.mark.parametrize('boxcoords',
+                         ['axes fraction', 'axes pixels', 'axes points',
+                          'data'])
+def test_picking(child_type, boxcoords):
+    # These all take up approximately the same area.
+    if child_type == 'draw':
+        picking_child = DrawingArea(5, 5)
+        picking_child.add_artist(mpatches.Rectangle((0, 0), 5, 5, linewidth=0))
+    elif child_type == 'image':
+        im = np.ones((5, 5))
+        im[2, 2] = 0
+        picking_child = OffsetImage(im)
+    elif child_type == 'text':
+        picking_child = TextArea('\N{Black Square}', textprops={'fontsize': 5})
+    else:
+        assert False, f'Unknown picking child type {child_type}'
+
+    fig, ax = plt.subplots()
+    ab = AnnotationBbox(picking_child, (0.5, 0.5), boxcoords=boxcoords)
+    ab.set_picker(True)
+    ax.add_artist(ab)
+
+    calls = 0
+    def on_pick(event):
+        nonlocal calls
+        if event.artist == ab:
+            calls += 1
+    fig.canvas.mpl_connect('pick_event', on_pick)
```

Comment:
perhaps
```
calls = []  # or "events"
fig.canvas.mpl_connect("pick_event", calls.append)
```
and later
```
calls.clear()
assert len(calls) == 1 and calls[0].artist == ab
```
?

Reply:
Unfortunately, this doesn't work because `mpl_connect` tries to take a `cbook._StrongRef`, and `list` is unhashable, but I've committed a slight variation.